python/config.cpp

```diff
@@ -1,9 +1,50 @@
-#include <pybind11/pybind11.h>
+#include <arbor/version.hpp>
+
+#include <sstream>
+#include <iomanip>
+#include <ios>
 
-#include "config.hpp"
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
 
 namespace pyarb {
+    
+// Create and return a dictionary
+pybind11::dict config() {
+    pybind11::dict dict;
+#ifdef ARB_MPI_ENABLED
+    dict[pybind11::str("mpi")]     = pybind11::bool_(true);
+#else
+    dict[pybind11::str("mpi")]     = pybind11::bool_(false);
+#endif
+#ifdef ARB_WITH_MPI4PY
+    dict[pybind11::str("mpi4py")]  = pybind11::bool_(true);
+#else
+    dict[pybind11::str("mpi4py")]  = pybind11::bool_(false);
+#endif
+#ifdef ARB_WITH_GPU
+    dict[pybind11::str("gpu")]     = pybind11::bool_(true);
+#else
+    dict[pybind11::str("gpu")]     = pybind11::bool_(false);
+#endif
+    dict[pybind11::str("version")] = pybind11::str(ARB_VERSION);
+    return dict;
+}
+
+void print_config(const pybind11::dict &d) {
+    std::stringstream s;
+    s << "Arbor's configuration:" << std::endl;
```

Comment:
just `<< Arbor's configuration:\n";
`std::endl` can have heavy performance cost, and should only be used when you really want to force that the output is flushed.

Reply:
Done